flag_engine/identities/models.py

```diff
@@ -28,20 +47,23 @@ class IdentityModel:
     identity_features: typing.List[FeatureStateModel] = field(default_factory=list)
     identity_traits: typing.List[TraitModel] = field(default_factory=list)
 
+    def get_identity_traits_as_dict(self) -> typing.Dict[str, TraitModel]:
+        return {trait.trait_key: trait for trait in self.identity_traits}
+
     def get_all_feature_states(
-        self, environment: EnvironmentModel
+        self, environment: EnvironmentModel, traits: typing.List[TraitModel] = None
     ) -> typing.List[FeatureStateModel]:
         all_feature_states = {fs.feature: fs for fs in environment.feature_states}
+        with override_identity_traits(self, traits or self.identity_traits) as _:
+            for segment_override in environment.segment_overrides:
+                feature_state = segment_override.feature_state
+                feature = feature_state.feature
+                if self.in_segment(segment_override.segment):
```

Comment:
Couldn't we flatten this using a `filter` statement? 

```python
for matching_override in filter(lambda segment_override: self.in_segment(segment_override.segment), environment.segment_overrides): 
    ...
```

We could perhaps even extract that lambda to a function to simplify it / make it more readable?